### PR TITLE
Multiple bugzilla 5 fixes

### PR DESCRIPTION
**Pull request by AdamWill**
This PR includes fixes for various issues I noted with how this thing behaves since the switch from querying Bugzilla to just using the data from the upstream Bugzilla 5 messages. It also adds a test suite! We all love tests, right?

### CODE REVIEW

**Comment by AdamWill** [comment]
@ralphbean @jfearn @pypingou

**Comment by AdamWill** [comment]
The errors look like tox is having trouble downloading/unpacking deps for some reason. The first time it errored on the Rawhide pipeline, I force-pushed a minor edit to make the tests run again and that time Rawhide worked fine and F29 errored. I don't think there's any actual bug in the PR.

**Comment by AdamWill** [comment]
Now rebased on the fedora-messaging migration.

**Review by ralphbean — APPROVED** [review]

**Comment by AdamWill** [comment]
Last change was just a minor edit to a comment in .coveragerc to make it consistent.

File: bugzilla2fedmsg/relay.py [hunk 1]
```diff
@@ -67,26 +66,48 @@ def __init__(self, config):
...
+        objdict = {}
+        bug = None
+        if obj == 'bug':
+            # just take the bug dict
+            bug = body['bug']
+        else:
+            # unpick the bug dict from the object dict
+            bug = dict(body[obj]['bug'])
+            objdict[obj] = dict(body[obj])
+            del(objdict[obj]['bug'])
```
Comment (abompard) [inline bugzilla2fedmsg/relay.py]: Dictionaries have a [pop()](https://docs.python.org/3/library/stdtypes.html#dict.pop) method that could be useful here. (just a FYI, not a required change)

**Review by abompard — APPROVED** [review]
This looks great, thanks!

File: bugzilla2fedmsg/relay.py (hunk 1 above)
Comment (AdamWill) [reply bugzilla2fedmsg/relay.py]: Yeah, that'd make it cleaner. One thing I'm doing there (which the comment doesn't make clear; bad me) is intentionally not changing the *original* dict; both the `bug` dict and the `objdict[obj]` dict are intentionally created as copies, and any modifications applied to those. I wanted to preserve the initial dict unmodified. Still, I'm not sure that's really necessary, and we could probably simplify this anyway...

**Comment by AdamWill** [comment]
Tweaked a bit in response to @abompard 's point, this is the diff against the previous version, for the record:

    [adamw@adam bugzilla2fedmsg (bz5-fixes)]$ git diff origin/bz5-fixes 
    diff --git a/bugzilla2fedmsg/relay.py b/bugzilla2fedmsg/relay.py
    index 4af22c8..8cc213e 100644
    --- a/bugzilla2fedmsg/relay.py
    +++ b/bugzilla2fedmsg/relay.py
    @@ -83,15 +83,12 @@ class MessageRelay:
             objdict = {}
             bug = None
             if obj == 'bug':
    -            # just take the bug dict
    -            bug = body['bug']
    +            # just take the bug dict, converting datetimes
    +            bug = convert_datetimes(body['bug'])
             else:
                 # unpick the bug dict from the object dict
    -            bug = dict(body[obj]['bug'])
    -            objdict[obj] = dict(body[obj])
    -            del(objdict[obj]['bug'])
    -            # run convert_datetimes on the dict
    -            objdict[obj] = convert_datetimes(objdict[obj])
    +            bug = convert_datetimes(body[obj].pop('bug'))
    +            objdict[obj] = convert_datetimes(body[obj])
     
             # As of https://bugzilla.redhat.com/show_bug.cgi?id=1248259, bugzilla
             # will send the product along with the initial message, so let's check
    @@ -103,7 +100,6 @@ class MessageRelay:
     
             LOGGER.debug("Organizing metadata for #%s" % bug["id"])
             bug = dict([(attr, bug.get(attr, None)) for attr in BUG_FIELDS])
    -        bug = convert_datetimes(bug)
     
             body["timestamp"] = datetime.datetime.fromtimestamp(
                 int(headers["timestamp"]) / 1000.0, pytz.UTC
    diff --git a/tox.ini b/tox.ini
    index 8781b3b..deb430e 100644
    --- a/tox.ini
    +++ b/tox.ini
    @@ -7,9 +7,8 @@ skip_missing_interpreters = True
     deps =
         pytest
         pytest-cov
    -    mock
    -    datetime
         pytz
    +    mock
     commands =
         python -m pytest {posargs}
     # When running in OpenShift you don't have a username, so expanduser

**Comment by abompard** [comment]
Thanks again Adam

**Pull request merged**